core/primitives/src/bandwidth_scheduler.rs

```diff
@@ -364,23 +346,27 @@ impl BandwidthSchedulerParams {
 
     fn calculate(
         max_shard_bandwidth: Bandwidth,
+        max_single_grant: Bandwidth,
         max_allowance: Bandwidth,
         max_base_bandwidth: Bandwidth,
         max_receipt_size: Bandwidth,
         num_shards: u64,
     ) -> BandwidthSchedulerParams {
-        if max_shard_bandwidth < max_receipt_size {
-            panic!(
-                "max_shard_bandwidth is smaller than max_receipt_size! ({} < {}).
-                Invalid configuration - it'll be impossible to send a maximum size receipt.",
-                max_shard_bandwidth, max_receipt_size
-            );
-        }
+        assert!(
+            max_single_grant >= max_receipt_size,
+            "A max_single_grant can't be lower than max_receipt_size - it'll be impossible to send a max size receipt"
+        );
+        assert!(
+            max_single_grant <= max_shard_bandwidth,
+            "A single grant must always be less than max_shard_bandwidth"
```

Comment:
Nit: Technically, the code says that it must not be greater (but can be equal)

```suggestion
            "A single grant must not be greater than max_shard_bandwidth"
```